### PR TITLE
Feat: Blog x Search

### PR DESCRIPTION
**Pull request by ItzNotABug**
## What does this PR do?

Parent PR -
* https://github.com/appwrite/website/pull/1152

## Test Plan

(Write your test plan here. If you changed any code, please provide us with clear instructions on how you verified your changes work.)

## Related PRs and Issues

(If this PR is related to any other PR or resolves any issue or related to any issue link all related PR and issues here.)

### Have you read the [Contributing Guidelines on issues](https://github.com/appwrite/appwrite/blob/master/CONTRIBUTING.md)?

(Write your answer here.)

### CODE REVIEW

**Review by TorstenDittmann — COMMENTED** [review]
Lets make some changes to how the search/filter state is handled, lets make the category and search part of the query param and do the data fetching on the `+page.ts` 👍🏻

File: src/routes/blog/[[page]]/+page.svelte [hunk 1]
```diff
@@ -34,6 +41,51 @@
         });
     });
 
+    let selectedCategory = $page.url.searchParams.get('category') ?? 'Latest';
+
+    const handleSearch = async (value: string) => {
```
Comment (ernstmul) [inline src/routes/blog/[[page]]/+page.svelte:46]: The argument here is always the value from the global query right? Why not make this function without an argument, and use the global `query`? Line 47 would then become something like `const queryLowercase = query.toLowerCase()`

File: src/routes/blog/[[page]]/+page.svelte [hunk 2]
```diff
@@ -316,4 +468,113 @@
...
+    }
+
+    .is-selected {
+        color: hsl(var(--web-color-black));
+        background: hsl(var(--web-color-white));
+    }
+
+    .overflow-x-auto {
+        scrollbar-width: none;
+        -ms-overflow-style: none;
```
Comment (ernstmul) [inline src/routes/blog/[[page]]/+page.svelte:483]: I'm not too sure about our support levels, but don't think this is needed anymore right? Current Edge should pick

File: src/routes/blog/[[page]]/+page.svelte [hunk 3]
```diff
@@ -316,4 +468,113 @@
...
+            max-width: 62.5%;
+        }
+    }
+
+    @media (max-width: 768px) {
+        .search-and-categories {
+            flex-wrap: wrap;
+            display: flex !important;
+            padding-block-start: 2rem;
+            flex-direction: column-reverse !important;
```
Comment (ernstmul) [inline src/routes/blog/[[page]]/+page.svelte:565]: Are these 2 importants really needed? Can you get rid of them by making the selector more specific?

File: src/routes/blog/[[page]]/+page.svelte [hunk 4]
```diff
@@ -316,4 +468,113 @@
...
+        .categories-wrapper {
+            max-width: 90vw;
+        }
+
+        .search-and-categories > .mobile-search-bar {
+            align-self: baseline;
+        }
+
+        .search-and-categories > .web-input-text-search-wrapper {
+            min-inline-size: unset !important;
```
Comment (ernstmul) [inline src/routes/blog/[[page]]/+page.svelte:577]: Same !important question as above

File: src/routes/blog/[[page]]/+page.svelte (hunk 1 above)
Comment (ItzNotABug) [reply src/routes/blog/[[page]]/+page.svelte:46]: ohh yeah that makes sense.

File: src/routes/blog/[[page]]/+page.svelte (hunk 2 above)
Comment (ItzNotABug) [reply src/routes/blog/[[page]]/+page.svelte:483]: I wasn't sure too either, added it just for safety. wouldn't want a broken view on some older edge maybe. wdyt? should I remove - `ms-*`?

File: src/routes/blog/[[page]]/+page.svelte (hunk 3 above)
Comment (ItzNotABug) [reply src/routes/blog/[[page]]/+page.svelte:565]: yes, we can remove those.

File: src/routes/blog/[[page]]/+page.svelte (hunk 4 above)
Comment (ItzNotABug) [reply src/routes/blog/[[page]]/+page.svelte:577]: I think we can remove this too.

File: src/routes/blog/[[page]]/+page.svelte (hunk 2 above)
Comment (ernstmul) [reply src/routes/blog/[[page]]/+page.svelte:483]: I double checked with Torsten, we can remove those!

File: src/routes/blog/[[page]]/+page.svelte [hunk 5]
```diff
@@ -34,6 +41,59 @@
...
+
+    const handleSearch = async () => {
+        const searchQuery = query.toLowerCase();
+
+        const url = new URL($page.url);
+        /**
+         * Navigate to the first page on search/filter to ensure consistent
+         * navigation experience when changing categories or search queries.
+         */
+        url.pathname = '/blog';
```
Comment (TorstenDittmann) [inline src/routes/blog/[[page]]/+page.svelte:54]:
```suggestion
        /**
         * Navigate to the first page on search/filter to ensure consistent
         * navigation experience when changing categories or search queries.
         */
        const url = new URL('/blog', $page.url);
```

File: src/routes/blog/[[page]]/+page.svelte [hunk 6]
```diff
@@ -246,28 +386,28 @@
...
-                    <FooterNav />
-                    <MainFooter />
-                </div>
+        </div>
+        <div>
+            <div class="container">
+                <FooterNav />
+                <MainFooter />
             </div>
         </div>
```
Comment (TorstenDittmann) [inline src/routes/blog/[[page]]/+page.svelte:412]: do we need the `div` around the `div.container`?

File: src/routes/blog/[[page]]/+page.svelte [hunk 7]
```diff
@@ -246,28 +386,28 @@
                             {#if data.currentPage < data.totalPages}
                                 <a
                                     data-sveltekit-noscroll
-                                    class="flex navigation-button"
+                                    class="navigation-button flex"
                                     href="/blog/{data.currentPage + 1}"
```
Comment (TorstenDittmann) [inline src/routes/blog/[[page]]/+page.svelte:390]: the pagination will remove `category` or `search` queries 👍🏻 we need to add those here.

feel free to create a helper function that builds a link

File: src/routes/blog/content.ts [hunk 8]
```diff
@@ -105,9 +106,16 @@ export const categories = Object.values(categoriesGlob).map((categoryList) => {
     };
 });
 
-export const getAllBlogEntriesWithAuthors = () => {
+export const normalizeCategory = (str: string) => str?.replace(/\s+/g, '-').toLowerCase();
```
Comment (TorstenDittmann) [inline src/routes/blog/content.ts:109]: this may be a bit sketchy but good for now, we need a proper category map at some point I think :-)

Comment (ItzNotABug) [reply src/routes/blog/content.ts:109]: we definitely do, there's quite inconsistency for categories mentioned and used across posts.

File: src/routes/blog/[[page]]/+page.svelte (hunk 6 above)
Comment (ItzNotABug) [reply src/routes/blog/[[page]]/+page.svelte:412]: yes, for the footer's container. else the footer container just squeezes in width.

File: src/routes/blog/[[page]]/+page.svelte (hunk 7 above)
Comment (ItzNotABug) [reply src/routes/blog/[[page]]/+page.svelte]: Done.

**Review by ernstmul — APPROVED** [review]

**Pull request merged**